### PR TITLE
Cut through red state

### PR DESCRIPTION
**Pull request by silasdavis**
Does the:

- Remove trivial binary helpers
- Move storage code only used by forensics to forensics
- Remove dead code
- Add static analysis in makefile

### CODE REVIEW

File: vent/service/rowbuilder.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func buildBlkData(tbls types.EventTables, block *exec.BlockExecution) (types.Eve
 	if _, ok := tbls[tables.Block]; ok {
 		blockHeader, err := json.Marshal(block.Header)
 		if err != nil {
-			return types.EventDataRow{}, fmt.Errorf("Couldn't marshal BlockHeader in block %v", block)
+			return types.EventDataRow{}, fmt.Errorf("couldn not marshal BlockHeader in block %v", block)
```
Comment (gregdhill) [inline vent/service/rowbuilder.go]: typo

File: storage/rwtree_test.go [hunk 2]
```diff
@@ -13,14 +13,15 @@ import (
...
+	gaa := []byte("gaa")
+	dam := []byte("dam")
 	rwt.Set(foo, gaa)
-	rwt.Save()
+	_, _, err := rwt.Save()
+	require.NoError(t, err)
 	assert.Equal(t, gaa, rwt.Get(foo))
 	rwt.Set(foo, dam)
-	_, _, err := rwt.Save()
+	_, _, err = rwt.Save()
```
Comment (gregdhill) [inline storage/rwtree_test.go]: Do you think it would be worth adding a check that the version has been incremented?

File: consensus/abci/app.go [hunk 3]
```diff
@@ -34,8 +34,6 @@ type App struct {
 	// State
 	blockchain              *bcm.Blockchain
 	validators              Validators
-	checkTx                 func(txBytes []byte) types.ResponseCheckTx
```
Comment (gregdhill) [inline consensus/abci/app.go]: Why are we removing these fields?

**Comment by gregdhill** [comment]
Looks good! I reckon we could also build more table driven tests in terms of storage, but that's a task for another day.

File: consensus/abci/app.go (hunk 3 above)
Comment (silasdavis) [reply consensus/abci/app.go]: unused

**Comment by gregdhill** [comment]
Merged to update later.

**Pull request merged**